### PR TITLE
uplink updates

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

Nothing to the players
Changes uplink shop to depend it on uplink type. It was dependent on game mode before so if we want to create both traitors and nuclear agents, we would fail
Also added all uplink items to admin uplink(Syndicate Officer or spawnable radio) all uplink items

## Why It's Good For The Game

More freedom to admins for making rounds more interesting and unexpectable. Nukie team after 100 mins of boring round? Go for it!

## Testing

Compiled and runned on current build, spawned random uplinks, summoned items

## Changelog
Nothing changed to the players

### CODE REVIEW

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -12,9 +12,9 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 			var/datum/uplink_item/I = new path
 			if(!I.item)
 				continue
-			if(I.gamemodes.len && SSticker && !(SSticker.mode.type in I.gamemodes))
+			if(I.uplinktypes.len && SSticker && !(U.uplink_type in I.uplinktypes) && U.uplink_type != UPLINK_TYPE_ADMIN)
```
Comment (Sirryan2002) [inline code/datums/uplink_item.dm:15]:
```suggestion
			if(length(I.uplinktypes) && !(U.uplink_type in I.uplinktypes) && U.uplink_type != UPLINK_TYPE_ADMIN)
```
SSticker will never be null

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -12,9 +12,9 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
 				continue
-			if(I.excludefrom.len && SSticker && (SSticker.mode.type in I.excludefrom))
+			if(I.excludefrom.len && SSticker && (U.uplink_type in I.excludefrom))
```
Comment (Sirryan2002) [inline code/datums/uplink_item.dm:17]:
```suggestion
			if(length(I.excludefrom) && (U.uplink_type in I.excludefrom))
```
SSticker will never be null

**Review by Sirryan2002 — COMMENTED** [review]
few things after first pass

**Review by Sirryan2002 — COMMENTED** [review]
Code is fine but before I throw down an approval, can you test the following and ensure the uplinks spawned through these methods work:
- Nuclear Agent Spawning at round-start
- Spawning SST through admin tools
- Spawning SIT through admin tools
- Giving characters Syndicate Admin outfits

I know they work through game panel spawning, but please doublecheck these cases.

**Comment by HMBGERDO** [comment]
- [X] Nuclear Agent Spawning at round-start
Selected nuclear emergency gamemode in lobby, started, all good

- [X] Spawning SST through admin tools
Spawned SST through admin secret panel. Success. SST outfit doesnt have uplinks but if they had, it should be nuclear shop. Watch /mob/living/carbon/human/proc/equip_syndicate_commando to check their outfit

- [X] Spawning SIT through admin tools
Spawned SIT through admin secret panel. Success. SIT implant with the same as default traitor shop exclude job items. In future we can add items only for SIT or only for traitors if we want to

- [X] Giving characters Syndicate Admin outfits
Spawned human. Giving Syndicate Officer outfit. 500 TC admin uplink with all uplink items in the game. 
Entered round as Syndicate Officer from lobby, the same result.

**Review by Sirryan2002 — APPROVED** [review]
This is a great change, thank you for doing the extra tests. Items available in uplinks really shouldn't be tied to the gamemode but rather the object type! LGTM

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**